project/apps/matching-service/src/matching.gateway.ts

```diff
@@ -0,0 +1,114 @@
+import { Inject, Logger } from '@nestjs/common';
+import { ClientProxy } from '@nestjs/microservices';
+import {
+  OnGatewayConnection,
+  OnGatewayDisconnect,
+  WebSocketGateway,
+  WebSocketServer,
+} from '@nestjs/websockets';
+import { parse } from 'cookie';
+import { firstValueFrom } from 'rxjs';
+import { Server, Socket } from 'socket.io';
+import { MatchRedis } from './db/match.redis';
+
+enum MatchEvent {
+  'MATCH_FOUND' = 'match_found',
+  'MATCH_REQUEST_EXPIRED' = 'match_request_expired',
+}
+
+@WebSocketGateway(8080, {
+  cors: {
+    origin: 'http://localhost:3000',
+    credentials: true,
+  },
+})
+export class MatchingGateway
+  implements OnGatewayConnection, OnGatewayDisconnect
+{
+  @WebSocketServer()
+  server: Server;
+  private readonly logger = new Logger(MatchingGateway.name);
+
+  constructor(
+    @Inject('AUTH_SERVICE') private readonly authServiceClient: ClientProxy,
+    private readonly matchRedis: MatchRedis,
+  ) {}
+
+  async handleConnection(client: Socket) {
+    this.logger.log(`Client connected: ${client.id}`);
+
+    const cookie = client.handshake.headers.cookie;
+
+    // Disconnect Client if no cookie provided
+    if (!cookie) {
+      client.disconnect();
+      return;
+    }
+
+    const cookies = parse(cookie);
+    const token = cookies['token'];
+    // Disconnect client if no token provided
+    if (!token) {
+      client.disconnect();
+      return;
+    }
+    try {
+      const data = await firstValueFrom(
+        this.authServiceClient.send({ cmd: 'verify' }, token),
+      );
+
+      if (!data) {
+        this.logger.log('User Data not found');
+        client.disconnect();
+        return;
+      }
+
+      // TODO: Implement refresh token logic
```

Comment:
if im not wrong, currently if access token is expired, it won't be sent in the header and only the refresh token will be sent. so logic can be changed to: if no access token, try to retrieve refresh token. if no refresh token, disconnect. if have, refresh and verify.

i think similar logic to the auth guard in #25. ill modify once it's merged 👍